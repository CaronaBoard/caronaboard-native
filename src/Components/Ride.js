/* @flow */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text, TouchableOpacity } from 'react-native'
import Styles from './Styles/RideStyles'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { RkCard } from 'react-native-ui-kitten'

export default class Ride extends Component {
  renderLine (subTitle, icon) {
    return (
      <View style={Styles.line} >
        <Icon name={icon} style={Styles.icon} />
        <Text rkCardText>{subTitle}</Text>
      </View>
    )
  }

  renderFlexible (flexible) {
    return flexible ? this.renderLine('Rota flex√≠vel', 'call-split') : <View />
  }

  render () {
    const {
        area,
        origin,
        destination,
        flexible,
        days,
        hours,
        name,
        formUrl
     } = this.props.ride

    return (
      <TouchableOpacity onPress={() => console.log('touched ' + formUrl)}>
        <View>
          <RkCard>
            <View rkCardContent>
              <Text rkCardTitle>{area}</Text>
              { this.renderLine(origin, 'radio-button-unchecked') }
              { this.renderLine(destination, 'radio-button-unchecked') }
              { this.renderFlexible(flexible) }
            </View>
            <View rkCardContent>
              { this.renderLine(days, 'today') }
              { this.renderLine(hours, 'schedule') }
              { this.renderLine(name, 'directions-car') }
            </View>
          </RkCard>
        </View>
      </TouchableOpacity>
    )
  }
}

Ride.propTypes = {
  ride: PropTypes.shape({
    area: PropTypes.string.isRequired,
    days: PropTypes.string.isRequired,
    destination: PropTypes.string.isRequired,
    formUrl: PropTypes.string.isRequired,
    hours: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    origin: PropTypes.string.isRequired,
    flexible: PropTypes.bool.isRequired
  }).isRequired
}
