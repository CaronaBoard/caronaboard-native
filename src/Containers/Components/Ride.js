/* @flow */

import React, { Component } from 'react'
import {
  View,
  Text,
  StyleSheet
} from 'react-native'

export default class Ride extends Component {
  render () {
    const {
        area,
        days,
        destination,
        flexible,
        formUrl,
        hours,
        name,
        origin
     } = this.props.ride

    return (
      <View style={styles.container}>
        <Text>{area}</Text>
        <Text>{days}</Text>
        <Text>{destination}</Text>
        <Text>{flexible}</Text>
        <Text>{formUrl}</Text>
        <Text>{hours}</Text>
        <Text>{name}</Text>
        <Text>{origin}</Text>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  }
})
