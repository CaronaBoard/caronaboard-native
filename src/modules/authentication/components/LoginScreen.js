import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkTextInput, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton } from '../../shared/components/index'
import { styles } from '../containers/styles/LoginScreenStyles'

export const INITIAL_STATE = {
  email: '',
  password: '',
  loading: false
}

export default class LoginV2 extends React.Component {
  constructor (props) {
    super(props)
    this.state = INITIAL_STATE
  }

  onButtonPress = () => {
    const { email, password } = this.state
    const { onButtonPress } = this.props
    try {
      onButtonPress(email, password)
    } catch (error) {
      console.log(error)
    }
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.alert

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/logo.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <RkTextInput onChangeText={(email) => { this.setState({email}) }} rkType='rounded' placeholder='Username' autoCapitalize='none' />
        <RkTextInput onChangeText={(password) => { this.setState({password}) }} rkType='rounded' placeholder='Password' secureTextEntry />

        <View style={styles.centralized}>
          <GradientButton
            style={styles.loginButton}
            rkType='large'
            text='LOGIN'
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer()}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
