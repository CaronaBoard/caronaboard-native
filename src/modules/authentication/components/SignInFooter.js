import React, { Component } from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import { RkButton, RkText } from 'react-native-ui-kitten'
import Styles from './styles/SignInFooterStyles'
import { screens } from '../../../configuration/navigation/Screens'

export class SignInFooter extends Component {
  static propTypes = {
    email: PropTypes.string,
    navigator: PropTypes.object.isRequired
  }

  navigateTo = screen => () => this.props.navigator.push({screen})

  render () {
    return (
      <View>
        <View style={Styles.textRow}>
          <RkButton
            key='forgot-password-button'
            rkType='clear'
            onPress={this.navigateTo(screens.forgotPassword.id)}
          >
            <RkText rkType='header6'>Forgot your password?</RkText>
          </RkButton>
        </View>
        <View style={Styles.textRow}>
          <RkText rkType='primary3'>Donâ€™t have an account?</RkText>
          <RkButton
            key='signup-button'
            rkType='clear'
            onPress={this.navigateTo(screens.signUp.id)}
          >
            <RkText rkType='header6'> Sign up now</RkText>
          </RkButton>
        </View>
      </View>
    )
  }
}
