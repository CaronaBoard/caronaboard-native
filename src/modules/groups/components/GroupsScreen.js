import React from 'react'
import { RkText } from 'react-native-ui-kitten'
import {
  ScrollView,
  View,
  TouchableOpacity
} from 'react-native'
import PropTypes from 'prop-types'

import { styles } from './styles/GroupsScreen.style'
import { GroupPropType } from '../types'
import { WarningScreen } from '../../shared/components/WarningScreen.style'

const hardCodedGroup = {
  'id': 'twpoa',
  'members': {},
  'name': 'ThoughtWorks POA'
}

export class GroupsScreen extends React.Component {
  static propTypes = {
    uid: PropTypes.string.isRequired,
    groups: PropTypes.arrayOf(GroupPropType).isRequired,
    joinGroup: PropTypes.func.isRequired,
    changeGroup: PropTypes.func.isRequired,
    activeGroup: PropTypes.oneOfType([GroupPropType, {}])
  }

  renderActiveGroup = (title, activeGroup) => {
    if (!activeGroup.name) {
      return null
    }

    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        <View style={styles.row} >
          <View
            style={styles.rowButton}
          >
            <RkText rkType='header6'>{activeGroup.name}</RkText>
          </View>
        </View>
      </View>
    )
  }

  renderGroupList = (title, group, onPress = () => null) => {
    return (
      <View style={styles.section}>
        <View style={[styles.row, styles.heading]}>
          <RkText rkType='primary header6'>{title}</RkText>
        </View>
        {group.map(group => {
          return (
            <View style={styles.row} key={group.id}>
              <TouchableOpacity
                style={styles.rowButton}
                onPress={() => onPress(group)}
              >
                <RkText rkType='header6'>{group.name}</RkText>
              </TouchableOpacity>
            </View>
          )
        })}
      </View>
    )
  }

  render () {
    const {groups, uid, joinGroup, changeGroup, activeGroup} = this.props
    const groupsYouBelong = [hardCodedGroup]

    groups.forEach(group => {
      if (group.members.hasOwnProperty(uid)) {
        groupsYouBelong.push(group)
      }
    })

    if (!uid) {
      return (
        <WarningScreen activeGroup={'Must have profile first'} />
      )
    }

    return (
      <ScrollView style={styles.container}>
        {this.renderActiveGroup('GRUPO ATIVO', activeGroup)}
        {this.renderGroupList('GRUPOS QUE VOCE PERTENCE', groupsYouBelong, changeGroup)}
        {this.renderGroupList('TODOS OS GRUPOS', groups, joinGroup)}
      </ScrollView>
    )
  }
}
