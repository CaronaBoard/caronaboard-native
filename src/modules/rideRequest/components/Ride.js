import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { RkCard } from 'react-native-ui-kitten'
import Icon from 'react-native-vector-icons/MaterialIcons'
import { View, Text, TouchableOpacity } from 'react-native'

import { RidePropType } from '../types'
import Styles from './styles/Ride.style'
import { LoadingSpinnerView } from '../../shared/components/LoadingSpinnerView'

export class Ride extends Component {
  static propTypes = {
    ride: RidePropType.isRequired,
    onPress: PropTypes.func.isRequired,
    icon: PropTypes.node
  }

  state = {
    loading: false
  }

  onPress = async (ride) => {
    this.setState({loading: true})
    await this.props.onPress(ride)
    this.setState({loading: false})
  }

  renderLine (subTitle, icon) {
    return (
      <View style={Styles.line} >
        <Icon name={icon} style={Styles.icon} />
        <Text rkCardText>{subTitle}</Text>
      </View>
    )
  }

  renderIcon = () => {
    const { icon } = this.props
    return icon || <View />
  }

  render () {
    const { ride } = this.props
    const {
      origin,
      destination,
      days,
      hours,
      profile
    } = ride

    return (
      <LoadingSpinnerView isLoading={this.state.loading} >
        <TouchableOpacity onPress={() => this.onPress({ ride })}>
          <RkCard>
            <View rkCardContent>
              <Text rkCardTitle>{destination}</Text>
              { this.renderLine(origin, 'radio-button-unchecked') }
              { this.renderLine(destination, 'radio-button-unchecked') }
            </View>
            <View rkCardContent>
              { this.renderLine(days, 'today') }
              { this.renderLine(hours, 'schedule') }
              { this.renderLine(profile.name, 'directions-car') }
            </View>
            {this.renderIcon()}
          </RkCard>
        </TouchableOpacity>
      </LoadingSpinnerView>
    )
  }
}
