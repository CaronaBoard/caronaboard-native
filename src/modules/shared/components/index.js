export { Button } from './Button'
export { TextInput } from './TextInput'
export { GradientButton } from './GradientButton'
export { LinearGradient } from './LinearGradient'
