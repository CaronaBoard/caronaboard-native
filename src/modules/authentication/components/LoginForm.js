import React from 'react'
import { View, Image, Keyboard } from 'react-native'
import Toast, { DURATION } from 'react-native-easy-toast'
import { RkText, RkAvoidKeyboard } from 'react-native-ui-kitten'
import { GradientButton, TextInput } from '../../shared/components'
import { styles } from '../containers/styles/LoginScreenStyles'

export class LoginForm extends React.Component {
  static defaultProps = {
    footer: () => {},
    toast: {
      showAlert: false,
      message: ''
    }
  }

  state = {
    email: '',
    password: '',
    loading: false
  }

  onButtonPress = () => {
    const { email, password } = this.state
    this.props.onButtonPress(email, password)
  }

  componentWillReceiveProps (nextProps) {
    const { showAlert, message } = nextProps.toast

    if (showAlert) {
      this.refs.toast.show(message, DURATION.LENGTH_LONG)
    }
  }

  render () {
    return (
      <RkAvoidKeyboard
        style={styles.screen}
        onStartShouldSetResponder={e => true}
        onResponderRelease={e => Keyboard.dismiss()}>

        <View style={styles.header}>
          <Image style={styles.image} source={require('../../../assets/images/caronaboard-logo-azul.png')} />
          <RkText rkType='logo h0'>Caronaboard</RkText>
          <RkText rkType='light h1'>Awesome Slogan</RkText>
        </View>

        <TextInput
          onChangeText={(email) => { this.setState({email}) }}
          key='email-input'
          rkType='rounded'
          placeholder='Username'
          autoCapitalize='none'
        />
        <TextInput
          onChangeText={(password) => { this.setState({password}) }}
          key='password-input'
          rkType='rounded'
          placeholder='Password'
          secureTextEntry
        />

        <View style={styles.centralized}>
          <GradientButton
            rkType='large'
            style={styles.loginButton}
            text={this.props.buttonText}
            onPress={this.onButtonPress}
          />
        </View>

        {this.props.footer(this.state.email)}
        <Toast position='top' ref='toast' />
      </RkAvoidKeyboard>
    )
  }
}
