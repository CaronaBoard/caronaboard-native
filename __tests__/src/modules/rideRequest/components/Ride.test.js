import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { Ride } from '../../../../../src/modules/rideRequest/components/Ride'
import { rideOfferFixture } from '../../../../resources/fixtures/ride/offer'

const props = {
  ride: rideOfferFixture,
  userId: 'kakaroto',
  onPress: jest.fn()
}

describe('<Ride />', () => {
  const wrapper = shallow(<Ride {...props} />)

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
