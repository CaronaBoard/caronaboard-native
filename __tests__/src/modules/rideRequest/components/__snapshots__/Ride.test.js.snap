// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ride /> Should have a snapshot 1`] = `
<LoadingSpinnerView
  isLoading={false}
>
  <TouchableOpacity
    activeOpacity={0.2}
    focusedOpacity={0.7}
    onPress={[Function]}
  >
    <RkCard>
      <View
        rkCardContent={true}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          rkCardTitle={true}
        >
          Bomfim
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="radio-button-unchecked"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            PUC
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="radio-button-unchecked"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            Bomfim
          </Text>
        </View>
      </View>
      <View
        rkCardContent={true}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="today"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            Seg-Sex
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="schedule"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            19h
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="directions-car"
            size={12}
            style={
              Object {
                "fontSize": 15,
                "marginRight": 5,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            rkCardText={true}
          >
            Eduardo Moroni
          </Text>
        </View>
      </View>
      <View />
    </RkCard>
  </TouchableOpacity>
</LoadingSpinnerView>
`;
